pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/AstImplUtil.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.ast;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+import net.sourceforge.pmd.Rule;
+import net.sourceforge.pmd.lang.ast.Node;
+import net.sourceforge.pmd.lang.java.typeresolution.TypeHelper;
+
+/**
+ * KEEP PRIVATE
+ * @author Clément Fournier
+ */
+final class AstImplUtil {
+
+    private static final List<String> UNUSED_RULES
+        = Arrays.asList("UnusedPrivateField", "UnusedLocalVariable", "UnusedPrivateMethod", "UnusedFormalParameter");
+
+    private static final List<String> SERIAL_RULES = Arrays.asList("BeanMembersShouldSerialize", "MissingSerialVersionUID");
```

Comment:
completely outside the scope of this PR, but to have the AST know about the rules is a level of coupling we should avoid…

Reply:
I'll add a TODO on https://github.com/pmd/pmd/wiki/PMD-7.0.0-Java